### ISSUE TITLE
Fix Log4J vulnerability

### ISSUE THREAD

**Issue by antolinos** [ispyb/ISPyB#538]
On Dec. 9, 2021, a remote code execution (RCE) vulnerability in Apache log4j 2 was identified being exploited in the wild. Public proof of concept (PoC) code was released and subsequent investigation revealed that exploitation was incredibly easy to perform. By submitting a specially crafted request to a vulnerable system, depending on how the system is configured, an attacker is able to instruct that system to download and subsequently execute a malicious payload. Due to the discovery of this exploit being so recent, there are still many servers, both on-premises and within cloud environments, that have yet to be patched. Like many high severity RCE exploits, thus far, massive scanning activity for CVE-2021-44228 has begun on the internet with the intent of seeking out and exploiting unpatched systems. We highly recommend that organizations upgrade to the latest version (2.15.0-rc2) of Apache log4j 2 for all systems.

Affected Version
Apache Log4j 2.x <= 2.15.0-rc1

**Comment by hormiai76** [comment]
Hi @antolinos 
But we are using an old version (1.2.17) and our networking engineer said it seems to be OK and not affected by this issue. Is it right?
Alberto Nardella

**Comment by antolinos** [comment]
@hormiai76 

You are totally right. We are not concerned by the new vulnerability but we are concerned by many more as Log4J end of live was in 2015 and vulnerabilities has been detected  and not fixed: https://logging.apache.org/log4j/1.2/

```
End of Life
On August 5, 2015 the Logging Services Project Management Committee announced that Log4j 1.x had reached end of life. For complete text of the announcement please see the Apache Blog. Users of Log4j 1 are recommended to upgrade to Apache Log4j 2.

Security Vulnerabilities
A security vulnerability, CVE-2019-17571 has been identified against Log4j 1. Log4j includes a SocketServer that accepts serialized log events and deserializes them without verifying whether the objects are allowed or not. This can provide an attack vector that can be expoited. Since Log4j 1 is no longer maintained this issue will not be fixed. Users are urged to upgrade to Log4j 2.
```

I did the fix with the upgrade to version 2.15.0 that is the recommended version, I'd like to give a try internally and see if everything works as expected and if it is the case, I'd probably recommend to take the MR (just to be on the safe side) but it is true that is not that crucial.
